compiler/rustc_metadata/src/creader.rs

```diff
@@ -791,9 +791,6 @@ impl<'a, 'tcx> CrateLoader<'a, 'tcx> {
         info!("loading profiler");
 
         let name = Symbol::intern(&self.sess.opts.unstable_opts.profiler_runtime);
-        if name == sym::profiler_builtins && attr::contains_name(&krate.attrs, sym::no_core) {
-            self.sess.emit_err(errors::ProfilerBuiltinsNeedsCore);
-        }
```

Comment:
I don't think we can remove this check.

I did some spelunking on why the check was introduced, [and this comment explains why](https://github.com/rust-lang/rust/pull/79958#issuecomment-743958675): when `profiler_builtins` is injected, it brings its dependencies into the build, which include the `core` crate. Now, if you link that into a crate that defines its own lang items, like a `#![no_core]` crate or `core` itself (like we're trying to do), you will see the duplicate lang item errors we've been seeing.

It's correct that this error doesn't apply to us anymore, since the `profiler_builtins` we're injecting doesn't depend on core, but users passing `-C instrument-coverage` on `#![no_core]` crates still need to see this error.

Reply:
What if we set a different name like `-Zprofiler-runtime=profiler_runtime_no_core` in the build step? This way the check gets bypassed because the name doesn't match. This would also not affect the regular code path, when the user passes `-Cinstrument-coverage`.

Reply:
That works too!